drgn_tools/task.py

```diff
@@ -185,6 +191,133 @@ def get_command(task: Object) -> str:
     return escape_ascii_string(task.comm.string_())
 
 
+def get_ppid(task: Object) -> int:
+    """
+    :returns: Parent PID of the task
+    """
+    return task.parent.pid.value_()
+
+
+class TaskRss(NamedTuple):
+    """
+    : Represent's a task's resident set size. See task_rss().
+    """
+
+    rss_file: int
+    rss_anon: int
+    rss_shmem: int
+
+    @property
+    def total(self) -> int:
+        return self.rss_file + self.rss_anon + self.rss_shmem
+
+
+def get_task_rss(task: Object, cache: Optional[dict] = None) -> TaskRss:
+    """
+    Return the task's resident set size (RSS) in pages
+
+    The task's RSS is the number of pages which are currently resident in
+    memory. The RSS values can be broken down into anonymous pages (not bound to
+    any file), file pages (those associated with memory mapped files), and
+    shared memory pages (those which aren't associated with on-disk files, but
+    belonging to shared memory mappings). This function returns a tuple
+    containing each category, but the common behavior is to use the "total"
+    value which sums them up.
+
+    :param task: ``struct task_struct *`` for which to compute RSS
+    :param cache: if provided, we can use this to cache the mapping of
+      "mm_struct" to RSS. This helps avoid re-computing the RSS value for
+      processes with many threads, but note that it could result in out of date
+      values on a live system.
+    :returns: the file, anon, and shmem page values
+    """
+    try:
+        mmptr = task.mm.value_()
+        if mmptr is None:
+            return TaskRss(0, 0, 0)
```

Comment:
`mmptr` can never be `None`, so this is dead code and can be removed. If `mmptr` is NULL, it will become the integer value zero when you call `.value_()`.